AIPscan/Reporter/helpers.py

```diff
@@ -30,25 +36,94 @@ def translate_headers(headers):
         fields.FIELD_AIP_NAME: "AIP Name",
         fields.FIELD_AIPS: "AIPs",
         fields.FIELD_AIP_SIZE: "AIP Size",
+        fields.FIELD_AIP_UUID: "AIP UUID",
         fields.FIELD_ALL_AIPS: "All AIPs",
         fields.FIELD_COUNT: "Count",
         fields.FIELD_CREATED_DATE: "Created Date",
+        fields.FIELD_DATE_START: "Start Date",
+        fields.FIELD_DATE_END: "End Date",
+        fields.FIELD_DURATION: "Duration",
         fields.FIELD_DERIVATIVE_COUNT: "Derivative Count",
+        fields.FIELD_DERIVATIVE_FILE: "Derivative File",
         fields.FIELD_DERIVATIVE_FORMAT: "Derivative Format",
         fields.FIELD_DERIVATIVE_UUID: "Derivative UUID",
         fields.FIELD_FILE_COUNT: "File Count",
         fields.FIELD_FILE_TYPE: "Type",
         fields.FIELD_FILENAME: "Filename",
         fields.FIELD_FORMAT: "Format",
         fields.FIELD_FORMATS: "Formats",
+        fields.FIELD_ID: "ID",
         fields.FIELD_NAME: "Name",
-        fields.FIELD_ORIGINAL_UUID: "Original UUID",
+        fields.FIELD_ORIGINAL_FILE: "Original File",
         fields.FIELD_ORIGINAL_FORMAT: "Original Format",
+        fields.FIELD_ORIGINAL_NAME: "Original Name",
+        fields.FIELD_ORIGINAL_PUID: "Original PUID",
+        fields.FIELD_ORIGINAL_UUID: "Original UUID",
+        fields.FIELD_ORIGINAL_VERSION: "Original Version",
         fields.FIELD_PUID: "PUID",
         fields.FIELD_RELATED_PAIRING: "Related Pairing",
         fields.FIELD_SIZE: "Size",
         fields.FIELD_STORAGE_NAME: "Storage Service Name",
         fields.FIELD_TRANSFER_NAME: "Transfer Name",
+        fields.FIELD_USER: "User",
         fields.FIELD_VERSION: "Version",
     }
     return [field_lookup.get(header, header) for header in headers]
+
+
+def _remove_primary_keys(dictionary):
+    """Remove AIPscan primary keys from dictionary."""
+    PK_FIELDS = (fields.FIELD_ID, fields.FIELD_AIP_ID)
+    for field in PK_FIELDS:
+        try:
+            dictionary.pop(field, None)
+        except AttributeError:
+            pass
+
+
+def format_size_for_csv(rows):
+    """Return data prepared for CSV file.
+
+    :param rows: Data assembled for tabular report (list of dicts)
+
+    :returns: rows with formatted size field (list of dicts)
+    """
+    for row in rows:
+        try:
+            row[fields.FIELD_SIZE] = filesizeformat(row[fields.FIELD_SIZE])
+        except KeyError:
+            pass
+    return rows
+
+
+def download_csv(headers, rows, filename="report.csv"):
+    """Write CSV and send it as an attachment.
+
+    :param headers: Row headers (list of str)
+    :param rows: Data to write to CSV, returned from Data endpoint (list of dicts)
+    :param filename: CSV filename (str)
+    """
+    string_io = StringIO()
+    writer = csv.writer(string_io)
+    writer.writerow(headers)
+    for row in rows:
+        _remove_primary_keys(row)
```

Comment:
Not opposed, just curious: Is there any reason a user might want to know the AIPScan primary keys? Could there be a world in which an exported CSV for be imported again later, like as a migration without having to re-ingest every AIP? And would the primary keys even matter?

Reply:
I was thinking that we'd omit them because the primary keys might be noise/confusing to users - they're only internal to AIPscan, not visible in the UI, and don't reflect the data in the stored AIPs.

Re-import is an interesting question but I think for now we'll likely want to stick with fetching information from the Storage Service to ensure that stored AIPs and information in AIPscan don't diverge.

Reply:
OK, sounds good!